### ISSUE TITLE
replace_all_text for pptx

### ISSUE THREAD

**Issue by sushilashenoy** [davidgohel/officer#293]
I'm interested in having a function similar to body_replace_all_text, but for pptx documents (as far as I can tell that function only works with docx).
I am interested in implementing this myself, but I wanted to inquire before I get started to confirm that this functionality

- doesn't already exist in another form that I haven't recognized.
- doesn't not exist because of a specific barrier to implementation

Based on scanning the code quickly it seems like I would need to add a function to the `slide` object in `ppt_classes.R` similar to the `replace_all_text` function defined for `docx_part` in `docx_part.R`, plus an exported function similar to `body_replace_all_text` in `docx_replace.R`

Basically I am trying to automate the creation of reports which include vector graphics (ruling out docx) in such a way that the formatting can be copied from a source document. While it's possible to create formatted text programmatically using officer I would prefer to use a find/replace type of template so that I don't have to mess with the formatting at all.

**Comment by davidgohel** [comment]
Hi, thanks for asking :)

It doesn't exist because I never took the time to look at that and I think it's complex. But if you try to and succeed I will be happy to merge your PR. Some users asked it so I am sure it would be a good new feature for users.

> automate the creation of reports which include vector graphics

Why are you talking about vector graphics? There is a PowerPoint limitation that make text and images mixing impossible in a shape. Could you show an image of what you are trying to have as a result? Maybe I will have inputs for you if I understand exactly what you are trying to achieve.

> Based on scanning the code quickly [...]

You are totally right! :)

**Comment by sushilashenoy** [comment]
Great, thanks.

I prefer to use vector graphics for several reasons - they are editable, they can be “zoomed” in on and printed at high resolution. I’m not sure if I am familiar with the PowerPoint limitation you are referring to. Can you elaborate? I will try to find or make an example I can share.

**Comment by davidgohel** [comment]
II don't know if you are aware but package [rvg](https://davidgohel.github.io/rvg/) is writing MS vector graphics in PowerPoint files.

What I am saying is I did not really understand what you want to do. But if your goal is to mix text and image wiithin a shape, this will not be possible. 

Can not this be done by replacing a shape (with its id) by a vector graphic? What is producing the vector graphics?

**Comment by sushilashenoy** [comment]
Okay, I think I understand. I am already using rvg to create vector graphics in the pptx documents. I don't intend or want to use this text replacement feature for that. In the same document there is also formatted text that I'd like to fill in programmatically with specific values. Does that make sense?

**Comment by sushilashenoy** [comment]
Got it working but don't have time today to add to the vignette or write tests so I will hold off on submitting the PR.

https://github.com/sushilashenoy/officer/commit/5182e4955b8b5c108f4f94b897760f93d084ffc0

**Comment by davidgohel** [comment]
So far, it looks perfect, many thanks.

**Comment by davidgohel** [comment]
I am closing this issue for now. Your PR will be welcome if you can submit it.

**Comment by aaronpayneppa** [comment]
@sushilashenoy @davidgohel Did this ever get added to the package? The 'replace_text_on_slide' function would be perfect for a project I am working on. I tested sushilashenoy's version and it seemed like there was a scoping issue with 'xml_type' function and 'doc' object.

**Comment by sushilashenoy** [comment]
No, unfortunately, other things got in the way so I never was able to finish this.

**Comment by davidgohel** [comment]
@sushilashenoy thanks anyway, your version could be later used as a base for implementing that feature
